zio/zsonio/ztests/first-test.yaml

```diff
@@ -0,0 +1,118 @@
+zql: "*"
+
+output-flags: -f zson
+
+input: |
+  #port=uint16
+  #zenum=string
+  #flow=record[orig_h:ip,orig_p:port,resp_h:ip,resp_port:port]
+  #0:record[_path:string,ts:time,id:flow,proto:zenum]
+  0:[conn;1521911720.600725;[10.47.1.152;49562;23.217.103.245;80;]tcp;]
+  0:[conn;1521911720.6008;[10.128.0.207;13;10.47.19.254;14;]icmp;]
+  0:[conn;1521911720.60131;[10.128.0.207;59777;10.47.28.6;443;]tcp;]
+  #1:record[flows:array[flow]]
+  1:[[[10.47.1.152;49562;23.217.103.245;80;]]]
+  1:[[[10.47.1.152;49562;23.217.103.245;80;][10.128.0.207;13;10.47.19.254;14;]]]
+  #2:record[ips:set[ip]]
+  2:[[10.0.0.1;192.168.1.1;]]
+  #3:record[nums:set[uint64]]
+  3:[[1;2;3;]]
+  #4:record[u:union[uint64,string]]
+  4:[0:12;]
+  4:[1:foo;]
+  #5:record[m:map[string,int32]]
+  5:[[a;1;b;2;c;3;]]
+  5:[[d;5;e;6;]]
+
+output: |
+  {
+      _path: "conn",
+      ts: 2018-03-24T17:15:20.600725Z,
+      id: {
+          orig_h: 10.47.1.152,
+          orig_p: 49562 (uint16) (port),
+          resp_h: 23.217.103.245,
+          resp_port: 80 (port)
+      } (25) (flow),
+      proto: "tcp" (zenum)
+  } (24),
```

Comment:
I'm sorry if I missed the explanation, but why do these need a typeid next to them?

Reply:
If the body of a record is fully type-defined, then ZSON doezs not required the extra type decorator but it's also ok to have it.  The writer doesn't have logic to determine if it's redundant so it always includes it for now.  The reader (coming soon) will handle both cases correctly.